openedx/core/djangoapps/courseware_api/views.py

```diff
@@ -123,11 +104,14 @@ def get_object(self):
         overview.enrollment = {'mode': mode, 'is_active': is_active}
 
         overview.is_staff = has_access(self.request.user, 'staff', overview).has_access
-        overview.can_load_courseware = self._check_access(self.request.user, overview, overview.is_staff)
-
-        # TODO: TNL-7053 Legacy: Delete these two once ready to contract
-        overview.user_has_access = overview.can_load_courseware
-        overview.user_has_staff_access = overview.is_staff
+        overview.can_load_courseware = check_course_access(
```

Comment:
And here's the payoff - no custom _check_access function!  We just use `check_course_access` and we're done.